### DISCUSSION TITLE
Memory use different on both gpus using ddp

### DISCUSSION THREAD

**Discussion by surya-narayanan** [Lightning-AI/pytorch-lightning#11692] [DDP / multi-GPU / multi-node]
![image](https://user-images.githubusercontent.com/17240858/152006250-66d83657-74ab-403c-a0c9-57736db3ff74.png)

I have noticed that there is often a discrepancy between the memory allocated and use in multi-gpu training (allocated shown above, use shown below). Why is that? When the gradients are synced across gpus in step four of this checklist (https://pytorch-lightning.readthedocs.io/en/stable/advanced/multi_gpu.html#distributed-data-parallel), do they use one gpu more than another? 

![image](https://user-images.githubusercontent.com/17240858/152006967-3a542acf-5f71-4385-b2d2-06a5a78aab68.png)

Thanks for your time.
